File: Data Structures/Stack/LV2_12973.cpp
#include <iostream>
#include <string>
#include <stack>

using namespace std;

// 12:03 ~ 12:31 (30)

int solution(string s) {
    int answer = -1;

    stack <char> S;
    S.push(s[0]);
    for (int i = 1; i < s.size(); i++) {
        if (!S.empty() && S.top() == s[i]) {
            S.pop();
        }
        else {
            S.push(s[i]);
        }
    }

    answer = S.empty();


    return answer;
}

File: Data Structures/Stack/LV2_12973.py
def solution(s):
    stack = []
    for ch in s:
        if len(stack) == 0:
            stack.append(ch)
            continue
        if stack[-1] != ch:
            stack.append(ch)
        else:
            stack.pop()
    return 1 if len(stack) == 0 else 0